File: NeetCode/1.Array/Top K Frequent Elements.js
function top_k_Elem(nums, k) {
    
    let myDict = {};
    let valOfDict = [];
    
    for (let num of nums){
        if (!(num in myDict)){
            myDict[num] = 1
            
        } else {
            myDict[num] += 1
        }
    }
    for (let [key, value] of Object.entries(myDict)) {
        valOfDict.push(value)
    }
    let result = []
    valOfDict.sort()
    for (let val of valOfDict.slice(-2)){
        for (let [key, value] of Object.entries(myDict)){
            if (val == value){
                result.push(key)
            }
        }
    }
    console.log(result)
} 

let nums = [1,3,3,3,3,2,2,5,5,5,5]
let k = 2
top_k_Elem(nums, k)

File: NeetCode/1.Array/Top K Frequent Elements.py
def Frequent_Elements(nums, k):

    count = {}
    for num in nums:
        if num not in count:
            count[num] = 1
        else:
            count[num] += 1
    arr = []
    for key, val in count.items():
        arr.append([val, key])
    arr.sort()
    print(arr)
   
    res = []
    for i in range(k):
        res.append(arr.pop()[0])
    return res
    


nums = [1,3,3,3,3,2,2,5,5,5,5,5]
k = 2
Frequent_Elements(nums, k)

'''
    myDict = {}
    valuesOfDict = []
    for num in nums:
        if num not in myDict:
            myDict[num] = 1
        else:
            myDict[num] += 1
    
    for key, values in myDict.items():
        valuesOfDict.append(values)
    valuesOfDict.sort()

    result = []
    for value in valuesOfDict[-k:]:
        for key, values in myDict.items():
            if value == values:
                result.append(key)
    return result
'''